### ISSUE TITLE
Pixman installation fails on macOS

### ISSUE THREAD

**Issue by adamjstewart** [spack/spack#8596]
I am unable to build `pixman`.

### Expected Result

A successful installation.

### Actual Result
```
5 errors found in build log:
     594      CC       oob-test.o
     595      CC       infinite-loop.o
     596      CC       trap-crasher.o
     597      CC       fence-image-self-test.o
     598      CC       region-translate-test.o
     599      CC       fetch-test.o
  >> 600    utils-prng.c:207:27: error: use of unknown builtin '__builtin_shuffle' [-Wimplicit-function-declaration]
     601                randdata.vb = __builtin_shuffle (randdata.vb, bswap_shufflemask);
     602                              ^
  >> 603    utils-prng.c:207:25: error: assigning to 'uint8x16' (vector of 16 'uint8_t' values) from incompatible type 'int'
     604                randdata.vb = __builtin_shuffle (randdata.vb, bswap_shufflemask);
     605                            ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     606    2 errors generated.
  >> 607    make[2]: *** [utils-prng.lo] Error 1
     608    make[2]: *** Waiting for unfinished jobs....
     609      CC       a1-trap-test.o
  >> 610    make[1]: *** [all-recursive] Error 1
  >> 611    make: *** [all] Error 2
```
### Steps to reproduce the issue

```console
$ spack install pixman
```

### Information on your system

- macOS 10.13.5
- Clang 9.0.0

**Comment by s-sajid-ali** [comment]
This also fails with intel compiler on CentOS. Needed it for matplotlib+animation but gave up trying to fix it.

**Comment by alalazo** [comment]
@adamjstewart This is the same problem I faced using Intel compilers some time ago in #8449. There should be a comment in the code that explains that particular issue. 

Two notes:
1. Probably you can get by substituting the current check  with
```python
if '%intel' in spec or '%clang' in spec:
    ...
```
2. Looking around it seems this is a regression of #7911 which should be solved by #8034

**Comment by adamjstewart** [comment]
@alalazo Good spot! It sounds like #8034 should solve my problems. The problem is that my version of Clang updated to 9.1.0 but I never updated the `compilers.yaml` in Spack, so it still thinks it's 9.0.0.
